app/src/features/FhirResourceTree/TreeItem.tsx

```diff
@@ -44,33 +57,33 @@ const TreeItem = ({ elementNode }: TreeItemProps): JSX.Element => {
         break;
     }
   } else {
-    iconName = IconNames.LAYERS;
+    iconName =
+      elementNode.type === "Extension"
+        ? IconNames.CODE_BLOCK
+        : (iconName = IconNames.LAYERS);
   }
 
   return (
     <MuiTreeItem
       nodeId={elementNode.id}
+      classes={{ root: classes.root }}
       label={
-        <>
-          {iconName && <Icon icon={iconName} color="white" iconSize={15} />}
-          <Typography display="inline" color="textPrimary">
-            {elementNode.name}
-          </Typography>
-          <Typography
-            display="inline"
-            variant="subtitle2"
-            color="textSecondary"
-          >
-            {elementNode.type}
-          </Typography>
-        </>
+        <TreeItemLabel
+          iconName={iconName}
```

Comment:
There is no need to affect `iconName` in TreeItem anymore. Could you move all the relative code inside of `TreeItemLabel` component and get rid of the `iconName` props ?